src/test/java/com/google/cloud/spanner/pgadapter/JdbcMockServerTest.java

```diff
@@ -4966,6 +4965,11 @@ public void testUUIDParameter() throws SQLException {
     }
 
     assertEquals(1, mockSpanner.countRequestsOfType(ExecuteSqlRequest.class));
+    ExecuteSqlRequest request = mockSpanner.getRequestsOfType(ExecuteSqlRequest.class).get(0);
+    assertEquals(1, request.getParams().getFieldsCount());
+    // UUIDs should be sent to Spanner as untyped values, so Spanner can infer the correct data
+    // type. This allows clients to use UUID query parameters with both STRING and UUID columns.
+    assertEquals(0, request.getParamTypesCount());
```

Comment:
Should we have a test where we are binding UUID but querying string (varchar) column?

Reply:
Yes, that is what this test does: https://github.com/GoogleCloudPlatform/pgadapter/pull/3236/files#r2069324656

It is better to add such a test to an end-to-end test, as this also verifies that Spanner accepts the untyped value. The mock server would always accept it.